lib/runner.ts

```diff
@@ -325,8 +347,12 @@ export class Runner extends EventEmitter {
           }
 
           if (this.config_.restartBrowserBetweenTests) {
+            // TODO(sjelin): replace with warnings once `afterEach` support is required
             let restartDriver = () => {
-              browser_.restart();
+              if (!this.afterEachWasCalled) {
```

Comment:
Maybe if we name this boolean `frameworkUsesAfterEach` it would be clearer? I confused myself for a bit thinking we'd need to reset it to false (which we don't, obviously).